src/learning/ProfileInitializer.cpp

```diff
@@ -0,0 +1,214 @@
+  int catAboveCounter = 0;
+  int catBelowCounter = 0;
+
+  // candidates are computed thanks to the method
+  // ProfileInitializer::getProfilePerformanceCandidates(crit, catBelow,
+  // nbCategories);
+  for (Perf can : candidates) {
+    // if the performance of the candidate is higher than the profile
+    // performance and that the alternatives category is catAbove then we have
+    // correctly classified it
+    if (can.getValue() > imaginaryProfilePerformance and
+        altPerformance_.getAlternativeAssignment(can.getName()) == catAbove) {
+      ++catAboveCounter;
+      // if the performance of the candidate is lower than the profile
+      // performance and that the alternatives category is catBelow then we
+      //  have correctly classified it
+    } else if (can.getValue() < imaginaryProfilePerformance and
+               altPerformance_.getAlternativeAssignment(can.getName()) ==
+                   catBelow) {
+      ++catBelowCounter;
+    }
+  }
+  float proba;
+  proba = catAboveCounter / catFrequency[catAbove.getCategoryRank()] +
+          catBelowCounter / catFrequency[catBelow.getCategoryRank()];
+  return proba;
+}
+
+std::vector<Perf> ProfileInitializer::initializeProfilePerformance(
+    const Criterion &crit, Categories &categories,
+    const std::vector<float> &catFre) {
+  std::vector<float> categoryLimits;
+  int nbCategories = categories.getNumberCategories();
+
+  for (int i = 0; i < nbCategories - 1; i++) {
+    std::vector<float> altProba;
+    std::vector<Perf> candidates =
+        ProfileInitializer::getProfilePerformanceCandidates(crit, categories[i],
+                                                            nbCategories);
+    // OPTIM : POSSIBILITY parallelization synchrone
+    for (Perf cand : candidates) {
+      float proba = ProfileInitializer::weightedProbability(
+          cand, crit, categories[i], categories[i + 1], nbCategories, catFre,
+          candidates);
+      altProba.push_back(proba);
+    }
+
+    std::random_device rd;
+    float totProba = std::accumulate(altProba.begin(), altProba.end(), 0);
+    float randomNumber = getRandomUniformInt(rd(), 0, totProba);
+    float tmp = 0;
+    int index;
+    for (int i = 0; i < candidates.size(); i++) {
+      if (tmp < randomNumber) {
+        tmp += altProba[i];
+        index = i;
+      } else {
+        break;
+      }
+    }
+    categoryLimits.push_back(candidates[index].getValue());
+  }
+  std::reverse(categoryLimits.begin(), categoryLimits.end());
+  // this is a work around since we would actually need to construct a
+  // Performance with a categories object.
+  std::vector<Perf> vect_p;
+  for (int i = 0; i < categoryLimits.size(); i++) {
+    // Cannot give a nice name to it since each vector of Perf need to have
+    // same name
+    vect_p.push_back(
+        Perf(crit.getId(), "cat" + std::to_string(i), categoryLimits[i]));
+  }
+  return vect_p;
+}
+
+void ProfileInitializer::initializeProfiles(MRSortModel &model) {
```

Comment:
Just a note for later: memory wise this could be optimized by changing in place the values of the profiles instead of creating a new object.